fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/LaoDetailViewModel.java

```diff
@@ -141,7 +144,60 @@ protected void onCleared() {
         disposables.dispose();
     }
 
+    /**
+     * Sends a ElectionCastVotes message .
+     *
+     * <p>Publish a GeneralMessage containing ElectionCastVotes data.
+     *
+     * @param castVotes         the corresponding votes for that election
+     */
+    public void sendVote( CastVote castVotes) {
+        if (mCurrentElection.getValue() == null) {
+            Log.d(TAG, "failed to retrieve current election");
+            return;
+        }
+        Log.d(TAG, "sending a new vote in election : " + mCurrentElection.getValue() + " with election start time"+ mCurrentElection.getValue().getStartTimestamp());
```

Comment:
You could store mCurrentElection.getValue() in a variable election, to avoid repetitions